pkg/controllers/customcluster_controller.go

```diff
@@ -210,46 +233,167 @@ func (r *CustomClusterController) reconcile(ctx context.Context, customCluster *
 		return r.reconcileHandleProvisioning(ctx, customCluster, customMachine, cluster)
 	}
 
+	if phase == v1alpha1.ProvisionedPhase {
+		currentClusterInfo := getClusterInfoFromCustomMachine(customMachine)
+		provisionedClusterInfo, errGetProvision := r.getProvisionedClusterInfoFromConfigmap(ctx, customCluster)
+		if errGetProvision != nil {
+			log.Error(errGetProvision, "failed to get provisioned cluster Info from configmap")
+		}
+		// By comparing curClusterInfo and provisionedClusterInfo to decide whether to proceed reconcileScaleUp or reconcileScaleDown.
+		scaleUpWorkerNodes := findScaleUpWorkerNodes(provisionedClusterInfo.WorkerNodes, currentClusterInfo.WorkerNodes)
+		scaleDownWorkerNodes := findScaleDownWorkerNodes(provisionedClusterInfo.WorkerNodes, currentClusterInfo.WorkerNodes)
+		if len(scaleUpWorkerNodes) != 0 {
+			return r.reconcileScaleUp(ctx, customCluster, customMachine, cluster)
+		}
+		if len(scaleDownWorkerNodes) != 0 {
+			return r.reconcileScaleDown(ctx, customCluster, customMachine, cluster)
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if phase == v1alpha1.ScalingUpPhase {
+		return r.reconcileFinishScalingUp(ctx, customCluster, customMachine, cluster)
+	}
+
+	if phase == v1alpha1.ScalingDownPhase {
+		return r.reconcileFinishScalingDown(ctx, customCluster, customMachine, cluster)
+	}
+
 	return ctrl.Result{}, nil
 }
 
+// reconcileHandleScalingDown determine whether customCluster enter Provisioned phase or UnknownPhase phase when current phase is ScalingUp.
+func (r *CustomClusterController) reconcileFinishScalingUp(ctx context.Context, customCluster *v1alpha1.CustomCluster, customMachine *v1alpha1.CustomMachine, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	return r.reconcileUpdateStatusByCheckPodStatus(ctx, customCluster, customMachine, cluster, r.reconcileScaleUp, CustomClusterScaleUpAction, v1alpha1.UnknownPhase)
+}
+
+// reconcileHandleScalingDown determine whether customCluster enter Provisioned phase or UnknownPhase phase when current phase is ScalingDown.
+func (r *CustomClusterController) reconcileFinishScalingDown(ctx context.Context, customCluster *v1alpha1.CustomCluster, customMachine *v1alpha1.CustomMachine, cluster *clusterv1.Cluster) (ctrl.Result, error) {
+	return r.reconcileUpdateStatusByCheckPodStatus(ctx, customCluster, customMachine, cluster, r.reconcileScaleDown, CustomClusterScaleDownAction, v1alpha1.UnknownPhase)
```

Comment:
IIRC this is polling checking the worker pod status, and when it completed sucessfully, it ends, right?

So I would suggest moving the implement into `reconcileScaleUp`

Reply:
yes, ok, I will do it.